File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ingredient) {
    int cnt = 0;
    vector<int> v = {0};
    
    for (int x : ingredient) {
        if (v.back() == 1 && x == 2) v.back() = 12;
        else if (v.back() == 12 && x == 3) v.back() = 123;
        else if (v.back() == 123 && x == 1) {
            cnt++;
            v.pop_back();
        } else v.push_back(x);
    }
    
    return cnt;
}

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] ingredient) {
        ArrayList<Integer> list = new ArrayList<>();
        
        for (int i = 0; i < ingredient.length; i++) {
            list.add(ingredient[i]);
        }
        int cnt = 0;
        for (int i = 0; i < list.size(); i++) {
            if (i > 2) {
                if (list.get(i) == 1 && list.get(i-1) == 3 
                    && list.get(i-2) == 2 && list.get(i-3) == 1) {
                    cnt++;
                    list.remove(i);
                    list.remove(i-1);
                    list.remove(i-2);
                    list.remove(i-3);
                    i = i - 4;
                }
            }
        }
        return cnt;
    }
}